### ISSUE TITLE
Make "AEM-Rules-for-SonarQube" compatible with SonarQube 10.4 "DownloadOnlyWhenRequired" feature

### ISSUE THREAD

**Issue by agigleux** [Cognifide/AEM-Rules-for-SonarQube#240]
Hello,

I'm Alex, Product Manager at Sonar.

SonarQube 10.4 will come with a feature to only download Sonar analyzers and 3rd-party plugins when they are really required by the Scanner. To make this feature work, each analyzer or 3rd-party plugin should declare the list of languages on which they expect to raise issues through a new MANIFEST property called Plugin-RequiredForLanguages.

See https://community.sonarsource.com/t/the-sonarscanners-download-only-required-3rd-party-plugins/108156 for more details.

Regards

**Comment by dbrin** [comment]
This is now unfortunately blocking our 10.x upgrade

**Comment by tomasz-niedzwiedz-wttech** [comment]
Hi @agigleux, I think I misunderstood the semantics of the newly introduced property and thought this plugin wouldn't be affected.
We're not using the plugin these days so it never came up as an issue internally.

What needs to be done is an upgrade of the `sonar-packaging-maven-plugin`, followed by the declaration of `java` and `html` in the `requiredForLanguages` config, due to dependencies on the Sonar Java plugin and the Sonar HTML plugin. 

I've opened https://github.com/wttech/AEM-Rules-for-SonarQube/pull/242 to try and address this. There are a few more Maven changes due to incompatibilities between the current configuration in our POM and the new version of the packaging plugin. I'll try and have a stab at those later this week.

**Comment by dbrin** [comment]
Any update on this?

**Comment by wooshcz** [comment]
> I've opened #242 to try and address this. There are a few more Maven changes due to incompatibilities between the current configuration in our POM and the new version of the packaging plugin. I'll try and have a stab at those later this week.

Hi @tomasz-niedzwiedz-wttech , I did spend some time trying to understand the issue and I made several attempts to fix the maven build using the new version of `sonar-packaging-maven-plugin` in my fork. From what I understand is that the build finishes fine with `sonarLintSupported` set to `false`. However I am not able to tell if this plugin was intended to ever support SonarLint. The official documentation (https://docs.sonarsource.com/sonarqube/latest/extension-guide/developing-a-plugin/plugin-basics/) describes this property as follows:
> Whether the language plugin supports SonarLint or not. Only SonarSource analyzers and custom rules plugins for SonarSource analyzers should set this to true.

I now intend to test if the JAR file I've created in my fork (with SonarLint support disabled) is working with our SonarQube v10 installation. I'll report back with the results. Let me know if I can help you in any way.

Thanks!
Ondrej

**Comment by tomasz-niedzwiedz-wttech** [comment]
Thanks @wooshcz I'm afraid this plugin comes with custom rules for AEM code, as such, it was designed to run with SonarLint support. This allows developers to use the official IDE extensions to perform analyses using the custom rules.
I've got a lot on my plate at the moment so I've requested some help internally to have somoene look into this full time and see how we can make it work without throwing that capability away.
